### ISSUE TITLE
Cannot instantiate Collections defined as attributes

### ISSUE THREAD

**Issue by pjeanjean** [gemoc/ale-lang#14]
The following code generates an exception:
```java.lang.ClassCastException: The value of type 'class java.util.ArrayList' must be of type 'interface org.eclipse.emf.common.util.EList'```
```
behavior testale_behavior;

class ClassB {
	Sequence(testale::ClassA) seq;
}

open class ClassA {
	@main
	def void run() {
		testale_behavior::ClassB b := testale_behavior::ClassB.create();
		b.seq := Sequence{};
	}
}
```

As such, it is not currently possible to create attributes with a cardinality > 1 with ALE.

The exact same code works fine if `b` is a local variable instead.

**Comment by fcoulon** [comment]
Should be fixed by 128fc177f42a44abf3ddcf5ad2d27befee1bfa65

**Comment by pjeanjean** [comment]
Well, now it does work with attributes added to opened classes, but not with classes defined in ALE.
It looks like a similar patch needs to be applied to `caseFeatureAssignment()` in the class [MethodEvaluator](https://github.com/gemoc/ale-lang/blob/master/plugins/org.eclipse.emf.ecoretools.ale.core/src/org/eclipse/emf/ecoretools/ale/core/interpreter/MethodEvaluator.java)

**Comment by fcoulon** [comment]
Shoud be fixed by d129d26b538ef50259e432ecbacda6e7141cdc40
Can I close ?
